### PR TITLE
ORV2-3201 - Allow Staff to use date picker to select start date in past

### PR DESCRIPTION
**Pull request by glen-aot**
# Description

Allow staff users to select a past date in CustomDatePicker, display warning message when doing so.
Please see [onRouteBCSpecification](https://github.com/bcgov/onRouteBCSpecification/blob/dfdc6c7b558a4773063f71fc744e8cb07bdaeb61/Applying%20for%20Permits/Staff%20apply%20for%20permits.feature) for details

Fixes #[ORV2-3201](https://moti-imb.atlassian.net/browse/ORV2-3201?atlOrigin=eyJpIjoiZDIzNTEzMDAzN2U4NDM1ZjljYTg2MWUxY2Q1NGU5ODMiLCJwIjoiaiJ9)

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1711-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1711-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1711-dops.apps.silver.devops.gov.bc.ca/api)
- [Policy](https://onroutebc-1711-policy.apps.silver.devops.gov.bc.ca/api)
- [Scheduler](https://onroutebc-1711-scheduler.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1711-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1711-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1711-dops.apps.silver.devops.gov.bc.ca/api)
- [Policy](https://onroutebc-1711-policy.apps.silver.devops.gov.bc.ca/api)
- [Scheduler](https://onroutebc-1711-scheduler.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/src/features/permits/components/dashboard/ApplicationStepPage.tsx [hunk 1]
```diff
@@ -141,7 +138,12 @@ export const ApplicationStepPage = ({
...
   };
 
-  if (isInvalidApplication || !isValidApplicationStatus() || !companyId || !isPermitTypeAllowed()) {
+  if (
+    isInvalidApplication ||
+    !isValidApplicationStatus() ||
+    !companyId ||
+    !isPermitTypeAllowed()
+  ) {
     return <Navigate to={ERROR_ROUTES.UNEXPECTED} />;
```
Comment (krishnan-aot) [inline frontend/src/features/permits/components/dashboard/ApplicationStepPage.tsx]: Can you add a console statement?  makes life easier for debugging.
`console.error('The application cannot be displayed or edited');`

File: frontend/src/features/permits/pages/Application/ApplicationForm.tsx [hunk 2]
```diff
@@ -190,8 +190,7 @@ export const ApplicationForm = ({
   // When "Continue" button is clicked
   const onContinue = async (data: ApplicationFormData) => {
     const updatedViolations = await triggerPolicyValidation();
-    // prevent CV client continuing if there are policy engine validation errors
-    if (Object.keys(updatedViolations).length > 0 && !isStaffUser) {
+    if (Object.keys(updatedViolations).length > 0) {
```
Comment (krishnan-aot) [inline frontend/src/features/permits/pages/Application/ApplicationForm.tsx]: Isn't this the change you made recently? Is it correct to remove it here?

File: frontend/src/features/permits/components/dashboard/ApplicationStepPage.tsx (hunk 1 above)
Comment (glen-aot) [reply frontend/src/features/permits/components/dashboard/ApplicationStepPage.tsx]: good call, done

File: frontend/src/features/permits/pages/Application/ApplicationForm.tsx (hunk 2 above)
Comment (glen-aot) [reply frontend/src/features/permits/pages/Application/ApplicationForm.tsx]: that is correct, must have accidentally removed this when merging main - added back in

**Review by krishnan-aot — APPROVED** [review]

**Pull request merged**